container_registry/container_analysis/samples_test.py

```diff
@@ -0,0 +1,178 @@
+#!/bin/python
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from os import environ
+from time import sleep, time
+
+from google.api_core.exceptions import InvalidArgument
+from google.api_core.exceptions import NotFound
+from google.cloud.devtools.containeranalysis_v1alpha1.proto.\
+    containeranalysis_pb2 import Occurrence
+from google.cloud.devtools.containeranalysis_v1alpha1.proto.\
+    package_vulnerability_pb2 import VulnerabilityType
+from google.cloud.pubsub import SubscriberClient
+
+import samples
+
+PROJECT_ID = environ['GCLOUD_PROJECT']
+SLEEP_TIME = 1
+TRY_LIMIT = 20
+
+
+class TestContainerAnalysisSamples:
+
+    def setup_method(self, test_method):
+        print("SETUP " + test_method.__name__)
+        timestamp = str(int(time()))
+        self.note_id = "note-" + timestamp + test_method.__name__
+        self.image_url = "www." + timestamp + test_method.__name__ + ".com"
+        self.note_obj = samples.create_note(self.note_id, PROJECT_ID)
+
+    def teardown_method(self, test_method):
+        print("TEAR DOWN " + test_method.__name__)
+        try:
+            samples.delete_note(self.note_id, PROJECT_ID)
+        except NotFound:
+            pass
+
+    def test_create_note(self):
+        new_note = samples.get_note(self.note_id, PROJECT_ID)
+        assert new_note.name == self.note_obj.name
+
+    def test_delete_note(self):
+        samples.delete_note(self.note_id, PROJECT_ID)
+        try:
+            samples.get_note(self.note_obj, PROJECT_ID)
+        except InvalidArgument:
+            pass
+        else:
+            # didn't raise exception we expected
+            assert (False)
+
+    def test_update_note(self):
+        description = "updated"
+        self.note_obj.short_description = description
+        samples.update_note(self.note_obj, self.note_id, PROJECT_ID)
+
+        updated = samples.get_note(self.note_id, PROJECT_ID)
+        assert updated.short_description == description
+
+    def test_create_occurrence(self):
+        created = samples.create_occurrence(self.image_url,
```

Comment:
Prefer to avoid hanging indents (here and elsewhere):

    samples.create_occurrence(
            self.image_url,
            self.note_id,
            PROJECT_ID)